src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,96 @@
 package core.basesyntax;
 
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    private void createNewArray() {
+        int newSize = array.length + array.length / 2;
+        T[] newArray = (T[]) new Object[newSize];
+        System.arraycopy(array, 0, newArray, 0, newSize);
+        array = newArray;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == array.length) {
+            createNewArray();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
 
+        if (index < 0 && index > size) {
+            throw new ArrayListIndexOutOfBoundsException("This index no existing in this Array");
+        }
+        if (size == array.length) {
+            createNewArray();
+        }
+        array[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        int newSize = size + list.size();
+        T[] newArray = (T[]) new Object[newSize];
+        System.arraycopy(array, 0, newArray, 0, newSize);
+        array = newArray;
+        for (int i = 0; i > newSize; i++) {
```

Comment:
The loop condition is incorrect; it should be `i < list.size()`. The current condition will never be true, so the loop body will never execute.